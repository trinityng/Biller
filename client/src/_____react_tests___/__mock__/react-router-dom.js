import React from 'react';
// Just render plain div with its children
const rrd = require('react-router-dom');

rrd.BrowserRouter = ({children}) => <div>{children}</div>
module.exports = rrd;