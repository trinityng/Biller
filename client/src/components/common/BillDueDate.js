import React from 'react';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  CardFooter,
  Row,
  Col,
  Button,
} from 'shards-react';
import { DatePicker } from 'antd';
import moment from 'moment';
import update from 'immutability-helper';

const dateFormat = 'MM/DD';


class BillDueDate extends React.Component {
  constructor(){
    super();

    this.state = {
      title: 'Bill Due Dates',
      accountDueDates: [ 
        {
          name: 'Chase Test Credit',
          key: null,
          date: '12/2',
        },
        {
          name: 'Chase Test Credit VIP',
          key: null,
          date: '5/2',
        },
        {
          name: 'PG&E Test',
          key: null,
          date: '6/2',
        },
      ],
    }

    this.onDateChange = this.onDateChange.bind(this);
    this.onDateUpdate = this.onDateUpdate.bind(this);
  }

  componentDidMount () {

  }

  onDateChange = (index, dateString) => {
    console.log(index + " " + dateString);
    this.setState(update(this.state.accountDueDates[index], {date: {$set: dateString }}));
  }

  onDateUpdate = () => {
    console.log(this.state.accountDueDates);
  }

  render() { 
    return (
      <Card small>
        <CardHeader className="border-bottom">
          <h6 className="m-0">{this.state.title}</h6>
          <div className="block-handle" />
        </CardHeader>

        <CardBody className="p-0">
          <ListGroup small flush className="list-group-small">
            {this.state.accountDueDates.map((item, idx) => (
              <ListGroupItem key={idx} className="d-flex px-3">
                <span className="text-semibold text-fiord-blue">{item.name}</span>
                <span className="ml-auto text-right text-semibold text-reagent-gray">
                  <DatePicker 
                    style = {{ width: 100 }}
                    format = { dateFormat }
                    defaultValue = { moment(item.date, dateFormat) }       
                    allowClear = { false }
                    onChange = {( _dateObj , datestring) => this.onDateChange(idx, datestring)}
                  />
                </span>
              </ListGroupItem>
            ))}
          </ListGroup>
        </CardBody>

        <CardFooter className="border-top">
          <Row>
            {/* Update Button */}
            <Col>
              <Button theme="primary" className="mb-2 mr-1" onClick = {this.onDateUpdate}>
                Update
              </Button>
            </Col>

            {/* View All Due Dates */}
            <Col className="text-right view-report">
              {/* eslint-disable-next-line */}
              <a href="#">See all due dates &rarr;</a>
            </Col>
          </Row>
        </CardFooter>
      </Card>
    );  
  }
};

export default BillDueDate;
