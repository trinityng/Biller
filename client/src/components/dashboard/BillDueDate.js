import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  Card,
  CardHeader,
  CardBody,
  ListGroup,
  ListGroupItem,
  CardFooter,
  Badge,
  Row,
  Col,
} from 'shards-react';
import { DatePicker } from 'antd';
import moment from 'moment';
import { USER_TOKEN } from '../../constants';
import {
  changeBillDueDate,
  plaidBills,
  getNotification,
} from '../../api/plaid.api';

const dateFormat = 'MM-DD-YY';

class BillDueDate extends React.Component {
  onDateChange = (item, value) => {
    const token = localStorage.getItem(USER_TOKEN);
    const { email } = this.props.user;
    const { name } = item;

    const date =
      moment()
        .format('YYYY')
        .toString() +
      '-' +
      value.format('MM-DD').toString();

    this.props
      .changeBillDueDate({
        account_name: name,
        due_date: date,
        email: email,
        token: token,
      })
      .then(() => {
        const token = localStorage.getItem(USER_TOKEN);
        const user = { token, email };
        this.props.plaidBills(user);
        this.props.getNotification(user);
      });
  };

  render() {
    const { plaid } = this.props || {};
    const { bills } = plaid || [];

    return (
      <Card small className="h-100">
        <CardHeader className="border-bottom">
          <h6 className="m-0">Bill Due Dates</h6>
          <div className="block-handle" />
        </CardHeader>

        <CardBody className="p-0">
          <ListGroup small flush className="list-group-small">
            {bills && bills.length ? (
              bills.map((item, idx) => (
                <ListGroupItem key={idx} className="d-flex px-3">
                  <span className="text-semibold text-fiord-blue">
                    {item.name}
                    <br />
                    Amount:{' '}
                    {item.amount > 0 ? (
                      <Badge theme="danger" className="mb-0 mr-0">
                        ${item.amount}
                      </Badge>
                    ) : item.amount === 0 ? (
                      <Badge theme="success" className="mb-0 mr-0">
                        ${item.amount}
                      </Badge>
                    ) : (
                      <Badge theme="success" className="mb-0 mr-0">
                        -${item.amount * -1}
                      </Badge>
                    )}
                  </span>
                  <span className="ml-auto text-right text-semibold text-reagent-gray">
                    {item.due_date ? (
                      <DatePicker
                        showTime={{ format: 'HH:mm' }}
                        style={{ width: 120 }}
                        format={dateFormat}
                        defaultValue={moment(
                          moment(item.due_date, 'YYYY-MM-DD')
                            .format(dateFormat)
                            .toString(),
                          dateFormat
                        )}
                        allowClear={false}
                        disabled={item.amount <= 0 ? true : false}
                        onChange={this.onChange}
                        onOk={(item, value => this.onDateChange(item, value))}
                      />
                    ) : (
                      <DatePicker
                        showTime={{ format: 'HH:mm' }}
                        style={{ width: 120 }}
                        format={dateFormat}
                        allowClear={false}
                        disabled={item.amount <= 0 ? true : false}
                        onOk={(item, value => this.onDateChange(item, value))}
                      />
                    )}
                  </span>
                </ListGroupItem>
              ))
            ) : (
              <span />
            )}
          </ListGroup>
        </CardBody>

        <CardFooter className="border-top">
          <Row>
            {/* View All Due Dates */}
            <Col className="text-right view-report">
              {/* eslint-disable-next-line */}
              <a href="#">See all due dates &rarr;</a>
            </Col>
          </Row>
        </CardFooter>
      </Card>
    );
  }
}

// Store
function mapStateToProps(state) {
  return {
    user: state.user,
    plaid: state.plaid,
  };
}

function matchDispatchToProps(dispatch) {
  return bindActionCreators(
    { plaidBills, changeBillDueDate, getNotification },
    dispatch
  );
}

export default connect(mapStateToProps, matchDispatchToProps)(BillDueDate);
