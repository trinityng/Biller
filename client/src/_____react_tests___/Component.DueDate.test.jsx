import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux';
import {mount, shallow, configure } from 'enzyme';
import { MemoryRouter, Router } from "react-router-dom";
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';

import BillDueDate from '../components/dashboard/BillDueDate';


Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

configure({adapter: new Adapter()});

describe('BillDueDate components', () => {
    let initialState = {};
    let store;

    beforeEach(() => {
        store = mockStore(initialState)
    });

    it(' should render BillDueDate correctly', () => {
        const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter initialEntries={['/dashboard']}>
                <BillDueDate />
            </MemoryRouter>,
        </Provider>
        );

        expect(wrapper).toMatchSnapshot();
    });
});