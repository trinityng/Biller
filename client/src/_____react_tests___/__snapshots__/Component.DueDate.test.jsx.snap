// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BillDueDate components  should render BillDueDate correctly 1`] = `ReactWrapper {}`;
